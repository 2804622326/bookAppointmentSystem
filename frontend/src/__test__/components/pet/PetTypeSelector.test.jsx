import React from 'react';
import { render, waitFor, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import PetTypeSelector from '../../../components/pet/PetTypeSelector.jsx';

jest.mock('../../../components/pet/PetService.js', () => ({
  getAllPetTypes: jest.fn(() => Promise.resolve({ data: ['Dog'] }))
}));

test('loads types and handles change', async () => {
  const onChange = jest.fn();
  const { asFragment } = render(<PetTypeSelector value='' onChange={onChange} />);
  await waitFor(() => screen.getByText('Dog'));
  fireEvent.change(screen.getByRole('combobox'), { target: { value: 'Dog' } });
  expect(onChange).toHaveBeenCalled();
  expect(asFragment()).toMatchSnapshot();
});
