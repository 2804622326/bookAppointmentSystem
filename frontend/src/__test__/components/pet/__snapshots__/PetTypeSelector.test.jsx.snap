// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads types and handles change 1`] = `
<DocumentFragment>
  <div
    class="col"
  >
    <select
      class="form-control"
      id="petType"
      name="type"
      required=""
    >
      <option
        value=""
      >
        ...select type...
      </option>
      <option
        value="Dog"
      >
        Dog
      </option>
      <option
        value="add-new-item"
      >
        Add New
      </option>
    </select>
  </div>
</DocumentFragment>
`;
